app/routes/bookmarks/index.tsx

```diff
@@ -43,13 +44,17 @@ export default function BookmarksIndexRoute() {
           </Link>
         </div>
         <div style={{ flexGrow: "1", maxWidth: "500px" }}>
-          <Input
-            placeholder="Search bookmarks..."
-            type="search"
-            className="rounded-pill"
-            value={filterText}
-            onChange={(e) => setFilterText(e.currentTarget.value)}
-          />
+          <form onSubmit={(e) => e.preventDefault()} method="GET">
```

Comment:
If we have JS, preventDefault to use "search on keypress", otherwise if no JS, you'll type, then press enter to submit the form and filter